sentry-apache-http-client-5/src/main/java/io/sentry/transport/apache/ApacheHttpClientTransport.java

```diff
@@ -71,68 +85,105 @@ public void send(final @NotNull SentryEnvelope envelope, final @Nullable Map<Str
       final SentryEnvelope filteredEnvelope = rateLimiter.filter(envelope, sentrySdkHint);
 
       if (filteredEnvelope != null) {
-        currentlyRunning.increment();
-
-        try (final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-            final GZIPOutputStream gzip = new GZIPOutputStream(outputStream)) {
-          options.getSerializer().serialize(filteredEnvelope, gzip);
-
-          final SimpleHttpRequest request =
-              SimpleHttpRequests.post(requestDetails.getUrl().toString());
-          request.setBody(
-              outputStream.toByteArray(), ContentType.create("application/x-sentry-envelope"));
-          request.setHeader("Content-Encoding", "gzip");
-          request.setHeader("Accept", "application/json");
-
-          for (Map.Entry<String, String> header : requestDetails.getHeaders().entrySet()) {
-            request.setHeader(header.getKey(), header.getValue());
-          }
+        final SentryEnvelope envelopeWithClientReport =
+            clientReportRecorder.attachReportToEnvelope(filteredEnvelope, options);
+
+        if (envelopeWithClientReport != null) {
+          currentlyRunning.increment();
+
+          try (final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+              final GZIPOutputStream gzip = new GZIPOutputStream(outputStream)) {
+            options.getSerializer().serialize(envelopeWithClientReport, gzip);
+
+            final SimpleHttpRequest request =
+                SimpleHttpRequests.post(requestDetails.getUrl().toString());
+            request.setBody(
+                outputStream.toByteArray(), ContentType.create("application/x-sentry-envelope"));
+            request.setHeader("Content-Encoding", "gzip");
+            request.setHeader("Accept", "application/json");
+
+            for (Map.Entry<String, String> header : requestDetails.getHeaders().entrySet()) {
+              request.setHeader(header.getKey(), header.getValue());
+            }
+
+            if (options.getLogger().isEnabled(DEBUG)) {
+              options
+                  .getLogger()
+                  .log(DEBUG, "Currently running %d requests", currentlyRunning.getCount());
+            }
+
+            httpclient.execute(
+                request,
+                new FutureCallback<SimpleHttpResponse>() {
+                  @Override
+                  public void completed(SimpleHttpResponse response) {
+                    if (response.getCode() != 200) {
+                      options
+                          .getLogger()
+                          .log(ERROR, "Request failed, API returned %s", response.getCode());
+
+                      if (response.getCode() >= 500) {
```

Comment:
`h`: We need to ensure that not both the client SDKs and the server count `>= HTTP codes` to follow the rule 

> The one who drops an envelope item must record and report it.

Reply:
Asked on client infra channel, outcome here: https://github.com/getsentry/develop/pull/554

> SDKs can put a simple optional check for HTTP status codes in place where any code `>= 400` except `429` will be recorded as `network_error`.

Reply:
cc @denrase @mattjohnsonpint